LibreHardwareMonitorLib/Hardware/Battery/BatteryGroup.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+using LibreHardwareMonitor.Interop;
+
+namespace LibreHardwareMonitor.Hardware.Battery
+{
+    internal struct BatteryReportInfo
+    {
+        public string Name { get; set; }
+        public uint Tag { get; set; }
+        public SafeFileHandle BatteryHandle { get; set; }
+        public Kernel32.BATTERY_INFORMATION MoreInformation { get; set; }
+    }
```

Comment:
Thanks, instead of this, can we perhaps abstract some of the stuff in ctor and call that also in GetReport?

Reply:
Yes. We could also move those properties into the Battery class and call them from GetReport(). It will be necessary to add more battery properties. Properties would be updated in the Battery constructor and in the Update() method.